src/node-file-trace.ts

```diff
@@ -294,13 +294,34 @@ export class Job {
   }
 
   async emitDependency (path: string, parent?: string, filesToEmit?: FilesToEmit) {
-    if (this.processed.has(path)) return;
```

Comment:
This early return changed.

Do we need to consider the case when `this.processed.has(path)` is true but `this.emitDependencyCache.has(path)` is false?

Reply:
That shouldn't happen as they are both set the first time `emitDependency` is called for a path, this check was more so for TypeScript initially, we can remove it if it makes it less clear I think since we're checking `cacheItem` in the inner if already.

Reply:
Updated in https://github.com/vercel/nft/pull/239/commits/00d1f5871680ca3b0a8a67d3484cecd04a474dbf